python_modules/libraries/dagster-gcp/dagster_gcp/solids.py

```diff
@@ -0,0 +1,302 @@
+        'priority',
+        'query_parameters',
+        'schema_update_options',
+        'use_legacy_sql',
+        'use_query_cache',
+        'write_disposition',
+    ]
+    kwargs = {k: cfg.get(k) for k in fields if cfg.get(k) is not None}
+
+    if destination_encryption_configuration is not None:
+        kwargs['destination_encryption_configuration'] = EncryptionConfiguration(
+            kms_key_name=destination_encryption_configuration
+        )
+
+    if time_partitioning is not None:
+        kwargs['time_partitioning'] = TimePartitioning(
+            field=time_partitioning['field'],
+            expiration_ms=time_partitioning['expiration_ms'],
+            require_partition_filter=time_partitioning['require_partition_filter'],
+        )
+    return QueryJobConfig(**kwargs) if kwargs else None
+
+
+def _extract_load_job_config(cfg):
+    destination_encryption_configuration = cfg.get('destination_encryption_configuration')
+    time_partitioning = cfg.get('time_partitioning')
+
+    fields = [
+        'allow_jagged_rows',
+        'allow_quoted_newlines',
+        'autodetect',
+        'clustering_fields',
+        'create_disposition',
+        'destination_encryption_configuration',
+        'destination_table_description',
+        'destination_table_friendly_name',
+        'encoding',
+        'field_delimiter',
+        'ignore_unknown_values',
+        'max_bad_records',
+        'null_marker',
+        'quote_character',
+        'schema_update_options',
+        'skip_leading_rows',
+        'source_format',
+        'time_partitioning',
+        'use_avro_logical_types',
+        'write_disposition',
+    ]
+    kwargs = {k: cfg.get(k) for k in fields if cfg.get(k) is not None}
+
+    if destination_encryption_configuration is not None:
+        kwargs['destination_encryption_configuration'] = EncryptionConfiguration(
+            kms_key_name=destination_encryption_configuration
+        )
+
+    if time_partitioning is not None:
+        kwargs['time_partitioning'] = TimePartitioning(
+            field=time_partitioning['field'],
+            expiration_ms=time_partitioning['expiration_ms'],
+            require_partition_filter=time_partitioning['require_partition_filter'],
+        )
+    return LoadJobConfig(**kwargs) if kwargs else None
+
+
+class BigQuerySolidDefinition(SolidDefinition):
+    """
+    Executes BigQuery SQL queries.
+    """
+
+    def __init__(self, name, sql_queries, description=None):
+        name = check.str_param(name, 'name')
+        sql_queries = check.list_param(sql_queries, 'sql queries', of_type=str)
+        description = check.opt_str_param(description, 'description', 'BigQuery query')
+
+        def _transform_fn(context, _):
+            (project, location) = [context.solid_config.get(k) for k in ('project', 'location')]
+            query_job_config = context.solid_config.get('query_job_config', {})
+
+            client = bigquery.Client(project=project, location=location)
```

Comment:
Yeah, I steered away from modeling the client as a resource because of this: https://stackoverflow.com/questions/50838525/bad-request-error-while-querying-data-from-bigquery-in-a-loop/50843371

When you construct the job config, you can either associate it with the client: `bigquery.Client(..., default_query_job_config=cfg)` 

or with the query: 

`client.query(..., job_config=cfg`)

However, if you do the former, the client is not reusable because it tries to sink query data into the same anonymous destination table in BQ. So, the config needs to be solid-level vs. resource level to support this, and consequently a BQ resource in this regime would be just a wrapper over the one-liner client instantiation, which isn't that useful

Reply:
The resource could just be responsible for creating the client every time. That would be abstracted away from the user